gtk/src/auto/file_filter.rs

```diff
@@ -34,10 +36,40 @@ impl FileFilter {
         }
     }
 
-    //#[doc(alias = "gtk_file_filter_add_custom")]
-    //pub fn add_custom(&self, needed: FileFilterFlags, func: /*Unimplemented*/Fn(/*Ignored*/FileFilterInfo) -> bool, data: /*Unimplemented*/Option<Fundamental: Pointer>) {
-    //    unsafe { TODO: call ffi:gtk_file_filter_add_custom() }
-    //}
+    #[doc(alias = "gtk_file_filter_add_custom")]
+    pub fn add_custom<P: Fn(&FileFilterInfo) -> bool + 'static>(
+        &self,
+        needed: FileFilterFlags,
+        func: P,
+    ) {
+        let func_data: Box_<P> = Box_::new(func);
+        unsafe extern "C" fn func_func<P: Fn(&FileFilterInfo) -> bool + 'static>(
+            filter_info: *const ffi::GtkFileFilterInfo,
+            data: glib::ffi::gpointer,
+        ) -> glib::ffi::gboolean {
+            let filter_info = from_glib_borrow(filter_info);
```

Comment:
```suggestion
            let filter_info = &*(filter_info as *const FileFilterInfo);
```

Reply:
These bits are autogenerated; won't any changes get overwritten the next time someone runs the generator script?